pkgs/applications/misc/squeekboard/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, cmake
+, meson
+, ninja
+, pkgconfig
+, gnome3
+, glib
+, gtk3
+, wayland
+, wayland-protocols
+, rustc
+, cargo
+, libcroco
+, libxml2
+, libxkbcommon
+, rustPlatform
+, makeWrapper
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "squeekboard";
+  version = "1.9.1";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1nrihpffr91p3ipif5k5z9vcvkwxk6qy8hp93w13s7gnwl2g93hx";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
+    rustc
+    cargo
+    glib.dev
+    gtk3.dev
+    wayland
+    gnome3.gnome-desktop
+    makeWrapper
+  ];
+
+  buildInputs = [
+    glib.dev
+    gtk3.dev
+    gnome3.gnome-desktop
+    wayland
+    wayland-protocols
+    libxml2
+    libxkbcommon
+  ];
+
+  patches = [
+    ./rm-dep-libcroco.patch
+  ];
+
+  cargoSha256 = "1fkhj4i2l2hdk9wvld6ryvnm1mxfwx3s555r7n42pg9f5namn1sr";
+
+  NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0";
+
+  configurePhase = "meson . _build/";
+  buildPhase = ''
+    ninja -C _build test
+  '';
+  installPhase = ''
+    runHook preInstall && DESTDIR="$out" ninja -C _build install && runHook postInstall
+  '';
+
+  postFixup = ''
+    mkdir -p $out/bin
+    cp _build/src/squeekboard $out/bin/
```

Comment:
There's a `$out/usr/local/share/applications/sm.puri.Squeekboard.desktop` generated with a non-absolute `Exec=squeekboard`, which probably needs to be fixed too.

The weird `/usr/local` stuff probably is due to this derivation not using the meson and ninja setup hooks.

Reply:
> There's a $out/usr/local/share/applications/sm.puri.Squeekboard.desktop generated with a non-absolute Exec=squeekboard, which probably needs to be fixed too.

I added a patch to substitute `Exec=squeekboard` 

> The weird /usr/local stuff probably is due to this derivation not using the meson and ninja setup hooks.

I'm trying to make it work without customizing the build/install phases